### ISSUE TITLE
Upload Progress called only once with fraction value 1.0 before the upload request complete

### ISSUE THREAD

**Issue by MohammadAlBarari** [Alamofire/Alamofire#3813]
## What did you do?
I tried to track the upload progress with this code
  ```
  func upload<T: Decodable>(_ endpoint: Endpoint, withName: String, fileName: String, fileData: Data, progressBlock: @escaping ((Progress) -> Void)) -> AnyPublisher<Result<T, NetworkError>, Never> {
        NetworkManager.authenticator.endPoint = endpoint

        return session.upload(multipartFormData: { multipartFormData in
            multipartFormData.append(fileData, withName: withName, fileName: fileName)
        }, to: endpoint.url,
        method: endpoint.httpMethod,
        headers: HTTPHeaders(endpoint.headers ?? [:]))
            .customValidate(url: endpoint.url)
            .cURLDescription { description in
                self.logRequest(endpoint, description)
            }.uploadProgress(queue: .main) { progress in
                // Here, you can handle progress updates
                print("progress \(progress)")
                progressBlock(progress)
            }
            .publishResponse(using: CustomResponseSerializable<T>(endpoint: endpoint))
            .map { result in
                result.value!
            }
            .eraseToAnyPublisher()
    }
```
## What did you expect to happen?
it should be called several times, with accurate progress.

## What happened instead?
The upload progress block was called only once before the upload request was complete
it printed:
progress <NSProgress: 0x60000045b880> : Parent: 0x0 (portion: 0) / Fraction completed: 1.0000 / Completed: 6910000 of 6910000  
progress 1.0

## Alamofire Environment

**Alamofire Version:** "5.6.4"
**Dependency Manager:** cocoapods
**Xcode Version:** latest
**Swift Version:** latest
**Platform(s) Running Alamofire:**
**macOS Version Running Xcode:**

## Demo Project

ℹ Please link to or upload a project we can download that reproduces the issue.

**Comment by jshier** [comment]
This has come up before, and there's not really anything we can do here, as Alamofire relies on `URLSession`'s automatic behavior for uploads. If you'd like to investigate, you can put a breakpoint in Alamofire's implementation of `urlSession(_ session: URLSession, task: URLSessionTask, didSendBodyData bytesSent: Int64, totalBytesSent: Int64, totalBytesExpectedToSend: Int64)` to see what values we're getting directly from session. If it doesn't match what Alamofire produces through the progress APIs let us know. But every time I've investigated it's been `URLSession` not returning progress updates.

**Comment by jshier** [comment]
If the investigation turns up an Alamofire bug, feel free to open another issue with the details.
